lib/saluki-components/src/sources/dogstatsd/mod.rs

```diff
@@ -394,6 +400,7 @@ pub struct DogStatsD {
     codec: DogstatsdCodec,
     context_resolvers: ContextResolvers,
     pre_filters: Arc<Vec<Box<dyn Filter + Send + Sync>>>,
+    additional_tags: Vec<String>,
```

Comment:
We should make this, and the subsequent fields for the listener/handler context, use `Arc<[String]>` instead, and then we can save on the allocations when cloning for each handler.

`Arc<[T]> implements `From<Vec<T>>` so it should be as simple as `additional_tags: self.additional_tags.clone().into()` up on line 359.